src/librustc/ty/maps/plumbing.rs

```diff
@@ -277,10 +252,43 @@ macro_rules! define_maps {
                 DepNode::new(tcx, $node(*key))
             }
 
+            fn try_get_lock(tcx: TyCtxt<'a, $tcx, 'lcx>,
+                            span: Span,
+                            key: &$K)
+                            -> Result<LockGuard<'a, QueryMap<$tcx, Self>>,
+                                      Result<($V, DepNodeIndex), CycleError<$tcx>>>
+            {
+                let lock = loop {
+                    let lock = tcx.maps.$name.borrow_mut();
+                    let job = if let Some(value) = lock.map.get(key) {
+                        match *value {
+                            QueryResult::Started(ref job) => Some(job.clone()),
+                            QueryResult::Complete(ref value) => {
+                                profq_msg!(tcx, ProfileQueriesMsg::CacheHit);
+                                return Err(Ok(((&value.value).clone(), value.index)));
+                            },
+                            QueryResult::Poisoned => FatalError.raise(),
+                        }
+                    } else {
+                        None
+                    };
+                    let job = if let Some(job) = job {
+                        job
+                    } else {
+                        break lock;
```

Comment:
Is there a reason why this line is not just `return Ok(lock);`? I think that would be clearer.

Reply:
Yeah, that would be better. The `break` is there because this used to be part of another function.

Reply:
I've fixed this.

Reply:
:+1: